addons/web/controllers/home.py

```diff
@@ -164,3 +164,19 @@ def health(self, db_server_status=False):
         headers = [('Content-Type', 'application/json'),
                    ('Cache-Control', 'no-store')]
         return request.make_response(data, headers, status=status)
+
+    @http.route(['/robots.txt'], type='http', auth="none")
+    def robots(self, **kwargs):
+        allowed_routes = self._get_allowed_robots_routes()
+        robots_content = ["User-agent: *", "Disallow: /"]
+        robots_content.extend(f"Allow: {route}" for route in allowed_routes)
+
+        return request.make_response("\n".join(robots_content), [('Content-Type', 'text/plain')])
+
+    def _get_allowed_robots_routes(self):
+        """Override this method to return a list of allowed routes.
+
+        :return: A list of URL paths that should be allowed by robots.txt
+              Examples: ['/social_instagram/', '/sitemap.xml', '/web/']
+        """
+        return []
```

Comment:
Oh wait we're *adding* it in 16.0? We probably shouldn't have touched anything here and just added it in 17.0 my bad :grimacing: 

I think no robots.txt implicitly means everything is allowed so this might break stuff, potentially.
I'll make a correction PR